rasa/core/test.py

```diff
@@ -351,6 +359,37 @@ def _clean_entity_results(
     return cleaned_entities
 
 
+def _intent_response_key_from_parsed_data(parsed: Dict[Text, Any]) -> Text:
+
+    base_intent = parsed.get(INTENT, {}).get(INTENT_NAME_KEY, {})
+
+    try:
+        # return only base intent if intent is not a retrieval intent
+        if (
+            base_intent
+            not in parsed[RESPONSE_SELECTOR][RESPONSE_SELECTOR_RETRIEVAL_INTENTS]
+        ):
+            return base_intent
```

Comment:
does this really need to be within the `try`?

Reply:
plus 1, doesn't seem that we should throw an exception here (because how should we recover from it?) Since we deal with `dict` here, we could use `.get()` method and specify the default value in case the `key` was not found.